src/mapping-v3-core.ts

```diff
@@ -1204,30 +1235,132 @@ function populateAllExistingMinterConfigurations<T>(
     return;
   }
   // Check the new minter filter for any pre-allowlisted minters and update Projects accordingly
-  let numProjectsWithMinters = minterFilterContract.getNumProjectsWithMinters();
+  const numProjectsWithMintersResult = minterFilterContract.try_getNumProjectsWithMinters();
+  if (numProjectsWithMintersResult.reverted) {
+    log.warning(
+      "[WARN] MinterFilter at address {} does not implement getNumProjectsWithMinters() function.",
+      [minterFilterContract._address.toHexString()]
+    );
+    return;
+  }
+  let numProjectsWithMinters = numProjectsWithMintersResult.value;
   for (
     let i = BigInt.fromI32(0);
     i.lt(numProjectsWithMinters);
     i = i.plus(BigInt.fromI32(1))
   ) {
-    let projectAndMinterInfo = minterFilterContract.getProjectAndMinterInfoAt(
+    const projectAndMinterInfoResult = minterFilterContract.try_getProjectAndMinterInfoAt(
       i
     );
-    let projectId = projectAndMinterInfo.value0;
-    let minterAddress = projectAndMinterInfo.value1;
+    if (projectAndMinterInfoResult.reverted) {
+      log.warning(
+        "[WARN] MinterFilter at address {} reverts getProjectAndMinterInfoAt(uint256) function for arg value {}.",
+        [i.toString()]
+      );
+      continue;
+    }
+    const projectAndMinterInfo = projectAndMinterInfoResult.value;
+    const projectId = projectAndMinterInfo.value0;
+    const minterAddress = projectAndMinterInfo.value1;
 
-    let project = Project.load(
+    const project = Project.load(
       generateContractSpecificId(contract._address, projectId)
     );
 
     if (project) {
-      loadOrCreateAndSetProjectMinterConfiguration(
-        project,
-        minterAddress,
-        timestamp
+      const minter = loadOrCreateMinter(minterAddress, timestamp);
+      loadOrCreateAndSetProjectMinterConfiguration(project, minter, timestamp);
+    }
+  }
+}
+
+/**
+ * Populate all project minter configurations from a given shared MinterFilter
+ * (e.g. IMinterFilterV1)
+ * @dev for legacy minter filters, use function
+ * legacyPopulateAllExistingMinterConfigurations
+ */
+function populateAllExistingMinterConfigurations<T>(
+  minterFilterContract: IMinterFilterV1,
+  contract: T,
+  timestamp: BigInt
+): void {
+  if (
+    !(
+      contract instanceof GenArt721CoreV3 ||
+      contract instanceof GenArt721CoreV3_Engine ||
+      contract instanceof GenArt721CoreV3_Engine_Flex
+    )
+  ) {
+    return;
```

Comment:
Nit: do we want to log a warning here before we return?

Reply:
We do this primarily because AS doesn't support union types, so we are stuck using generic type for contract. I would generally consider this as ~boilerplate, and if we were to ever call this function with a different type than the three items listed here, we should definitely be picking up in our test suite.

I think this pattern is followed many places in our handler codebase, so @jakerockland @yoshiwarab - what are your thoughts on if we should log a warning when type checking?

Reply:
I see I see – my general inclination would be that it would be good practice to log warning when typechecking, but doing so as a later cleanup type item down the road seems perfectly acceptable. 

If this is a common boilerplate issue, could be worth writing a light typechecker utility that abstracts this check+warning-log behavior.

Reply:
This all sounds good to me - I've added a polish task for this! https://app.asana.com/0/1203634050572702/1205096158411073/f